WeatherApp.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react';
+import { View, StyleSheet, ActivityIndicator } from 'react-native';
+import { useNavigation } from '@react-navigation/native';
+import { createDrawerNavigator } from '@react-navigation/drawer';
+import * as Location from 'expo-location';
+import useWeatherAPI from './useWeatherAPI';
+import CurrentWeatherScreen from './CurrentWeatherScreen';
+import ForecastScreen from './ForecastScreen';
+
+const Drawer = createDrawerNavigator();
+const API_KEY = 'd7e88c2243f84a3eb41183055231408';
+
+const WeatherApp = () => {
+  const navigation = useNavigation();
+  const [location, setLocation] = useState(null);
+
+  useEffect(() => {
+    (async () => {
+      try {
+        const { status } = await Location.requestForegroundPermissionsAsync();
+        if (status === 'granted') {
+          const location = await Location.getCurrentPositionAsync({});
+          setLocation({
+            latitude: location.coords.latitude,
+            longitude: location.coords.longitude,
+          });
+        }
+      } catch (error) {
+        console.error('Error getting location:', error);
+      }
+    })();
+  }, []);
+
+  const { data: currentWeatherData, loading: currentWeatherLoading } = useWeatherAPI(
+    location ? `/current.json?q=${location.latitude},${location.longitude}` : '/current.json?q=Warwick,RI', // Default to Warwick if the location fails
+    API_KEY
+  );
+
+  const { data: forecastData, loading: forecastLoading } = useWeatherAPI(
+    location ? `/forecast.json?q=${location.latitude},${location.longitude}&days=7` : '/forecast.json?q=Warwick,RI&days=7',
+    API_KEY
+  );
+
+  if (currentWeatherLoading || forecastLoading) {
+    return <LoadingIndicator />;
+  }
+
+  return (
+    <View style={styles.container}>
+      <Drawer.Navigator initialRouteName="CurrentWeather" screenOptions={{
```

Comment:
Things can start to get a bit hairy when you start embedding navigators within components. In your current app we have navigators in the following files:
1. App.tsx
2. WeatherApp.tsx
3. ForecastScreen.tsx

If we went another three screens deep with three more navigators, it would already become difficult to mange in my opinion.